src/types.js

```diff
@@ -92,6 +92,10 @@ export type Descriptor = {
 
   get?: UndefinedValue | CallableObjectValue | AbstractValue,
   set?: UndefinedValue | CallableObjectValue | AbstractValue,
+
+  joinCondition?: AbstractValue,
```

Comment:
Could you add a comment here that explains what combination of properties may be present? I guess that if `joinCondition` is present, then none of `writable`, `enumerable`, `configurable`, `value`, `get`, `set` may be present? And `descriptor1` and `descriptor2` may only be defined if `joinCondition` is present?

Would be nice if that kind of invariant would actually get enforced somewhere. I am a bit afraid that there's code out there today or tomorrow that ignores `joinCondition`.